File: src/network/packet/data/ServerDisconnectReason.php
<?php

namespace pocketcloud\cloud\network\packet\data;

use pocketcloud\cloud\util\misc\Writeable;
use pocketcloud\cloud\util\trait\EnumHelperTrait;

enum ServerDisconnectReason implements Writeable {
    use EnumHelperTrait;

    case CLOUD_SHUTDOWN;
    case SERVER_SHUTDOWN;

    public function getName(): string {
        return $this->name;
    }

    public function write(): string {
        return $this->name;
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/type/ServerDisconnectReason.java
package de.pocketcloud.cloud.network.packet.type;

import de.pocketcloud.cloud.util.Writable;

public enum ServerDisconnectReason implements Writable<String> {

    CLOUD_SHUTDOWN,
    SERVER_SHUTDOWN;

    @Override
    public String write() {
        return name();
    }
}